src/main/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilder.java

```diff
@@ -1,47 +1,151 @@
-    TablePointer conceptTablePointer = TablePointer.fromTableName("concept", tableNameGenerator);
-    TableVariable conceptTableVariable = TableVariable.forPrimary(conceptTablePointer);
-    FieldVariable nameFieldVariable = conceptTableVariable.makeFieldVariable("concept_name");
-    FieldVariable idFieldVariable = conceptTableVariable.makeFieldVariable("concept_id");
+      SnapshotBuilderDomainOption domainOption,
+      int conceptId,
+      TableNameGenerator tableNameGenerator,
+      CloudPlatformWrapper platform) {
+
+    // concept table and its fields concept_name and concept_id
+    var conceptTablePointer = TablePointer.fromTableName(CONCEPT, tableNameGenerator);
+    var conceptTableVariable = TableVariable.forPrimary(conceptTablePointer);
+    var nameFieldVariable = conceptTableVariable.makeFieldVariable(CONCEPT_NAME);
+    var idFieldVariable = conceptTableVariable.makeFieldVariable(CONCEPT_ID);
+
+    // concept_ancestor joined on concept.concept_id = ancestor_concept_id
+    var conceptAncestorTablePointer =
+        TablePointer.fromTableName(CONCEPT_ANCESTOR, tableNameGenerator);
+    var conceptAncestorTableVariable =
+        TableVariable.forJoined(conceptAncestorTablePointer, ANCESTOR_CONCEPT_ID, idFieldVariable);
+    var descendantIdFieldVariable =
+        conceptAncestorTableVariable.makeFieldVariable(DESCENDANT_CONCEPT_ID);
+
+    // domain specific occurrence table joined on concept_ancestor.descendant_concept_id =
+    // 'domain'_concept_id
+    var domainOccurrenceTablePointer =
+        TablePointer.fromTableName(domainOption.getTableName(), tableNameGenerator);
+    var domainOccurenceTableVariable =
+        TableVariable.forJoined(
+            domainOccurrenceTablePointer, domainOption.getColumnName(), descendantIdFieldVariable);
+
+    // COUNT(DISTINCT person_id)
+    var countFieldVariable =
+        new FieldVariable(
+            new FieldPointer(domainOccurrenceTablePointer, PERSON_ID, "COUNT"),
+            domainOccurenceTableVariable,
+            "count",
+            true);
+
+    List<SelectExpression> select = List.of(nameFieldVariable, idFieldVariable, countFieldVariable);
+
+    List<TableVariable> tables =
+        List.of(conceptTableVariable, conceptAncestorTableVariable, domainOccurenceTableVariable);
 
-    TablePointer ancestorTablePointer =
-        TablePointer.fromTableName("concept_ancestor", tableNameGenerator);
-    TableVariable ancestorTableVariable = TableVariable.forPrimary(ancestorTablePointer);
-    FieldVariable descendantFieldVariable =
-        ancestorTableVariable.makeFieldVariable("descendant_concept_id");
+    List<FieldVariable> groupBy = List.of(nameFieldVariable, idFieldVariable);
 
-    BinaryFilterVariable whereClause =
+    List<OrderByVariable> orderBy =
+        List.of(new OrderByVariable(nameFieldVariable, OrderByDirection.ASCENDING));
+
+    // ancestorTable is primary table for the subquery
```

Comment:
Can you refactor the subquery into its own method? I think that would make the code clearer. I think the method would look something like
```
Query createSubQuery(int conceptId, TableNameGenerator tableNameGenerator)
```